File: app/src/main/java/com/example/macarus0/walkiewalkie/data/WalkieDatabaseProvider.java
package com.example.macarus0.walkiewalkie.data;

import android.arch.persistence.room.Room;
import android.content.Context;


public class WalkieDatabaseProvider {

    private static final String dbName = "WalkieDatabase";

    private static WalkieDatabase sDb;

    public static WalkieDatabase getDatabase(Context context) {
        if(sDb == null) {
            sDb = createDatabase(context);
        }
        return sDb;
    }

    private static WalkieDatabase createDatabase(Context context) {
        return Room.databaseBuilder(context.getApplicationContext(), WalkieDatabase.class, dbName).build();
    }

}

File: app/src/main/java/com/example/macarus0/walkiewalkie/data/WalkieDatabaseProvider.kt
package com.example.macarus0.walkiewalkie.data

import android.content.Context
import androidx.room.Room


object WalkieDatabaseProvider {

    private const val dbName = "WalkieDatabase"

    private lateinit var sDb: WalkieDatabase

    fun getDatabase(context: Context): WalkieDatabase {
        if (sDb == null) {
            sDb = createDatabase(context)
        }
        return sDb
    }

    private fun createDatabase(context: Context): WalkieDatabase {
        return Room.databaseBuilder<WalkieDatabase>(context.applicationContext, WalkieDatabase::class.java, dbName).build()
    }

}
